src/widgets/SimpleTable.js

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable react/no-unused-prop-types */
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { StyleSheet, FlatList, View, Text } from 'react-native';
+import { ROW_BLUE, APP_FONT_FAMILY, SUSSOL_ORANGE } from '../globalStyles/index';
+import { generateUUID } from '../database';
+
+/**
+ * Simple table component only rendering scalar values in a cell.
+ * Required props
+ * data = [ { a: 1, b: 2}, {a: 3, b: 4}, .. ]
+ * columns = [
+ * { key: 'a', width: 1, title: 'A COLUMN' },
+ * { key: 'b', width: 1, title: 'B COLUMN' } ..
+ * ]
+ * title: Title of the table.
+ */
+export class SimpleTable extends React.PureComponent {
+  keyExtractor = () => generateUUID();
+
+  renderTitle = () => {
+    const { title } = this.props;
+    const { titleContainer, titleFont } = localStyles(this.props);
+    return (
+      <View style={titleContainer}>
+        <Text style={titleFont}>{title}</Text>
+      </View>
+    );
+  };
+
+  renderCell = ({ content, index, header, width }) => {
+    const { cell, cellFont } = localStyles({ ...this.props, index, header, width });
+    return (
+      <View style={cell} key={this.keyExtractor()}>
```

Comment:
Often in this context you can be more basic and get away with ```key={index}```. Unless the cell order might change. At that point a better thought out keyExtractor is probably sensible (and best chance through props). UUID is still not the right approach, as it'll be different every time you render the cells.

As explained [here](https://reactjs.org/docs/lists-and-keys.html#keys), they keys only have to be unique among its siblings in the DOM tree, and they should be stable if possible. If the key is the same as some parent row it shouldn't matter, so long as it's not the same as another cell in the same row.

Reply:
done here https://github.com/openmsupply/mobile/pull/872#discussion_r276085908